libcnb-data/src/buildpack_plan.rs

```diff
@@ -16,6 +16,17 @@ pub struct Entry {
     pub metadata: Table,
 }
 
+impl Entry {
+    /// Deserializes Metadata to a type T that implements Deserialize
+    pub fn metadata<'de, T>(&self) -> Result<T, toml::de::Error>
+    where
+        T: Deserialize<'de>,
+    {
+        // serde::de::Deserializer is not implemented for toml::map::Map, so need to clone() here
+        toml::Value::Table(self.metadata.clone()).try_into()
```

Comment:
@Malax not sure we can remove `clone()` here. :(

Reply:
Since `try_into` consumes `self` (which is `self.metadata` in the context of this function) and `fn metadata` only has a reference to `self`, this seems unlikely to be possible.

Nit: the comment is therefore slightly incorrect. We don't need to clone because `Deserialize` isn't implemented, we need to clone because `try_into` will consume the value that is a field of `self`.

Reply:
While that's true I got here because you can't do `serde::de::Deserialize::deserialize(self.metadata))`, you'll run into:

```
error[E0277]: the trait bound `toml::map::Map<std::string::String, Value>: bom::_::_serde::Deserializer<'_>` is not satisfied
  --> libcnb-data/src/buildpack_plan.rs:24:9                                                                                             
   |                                                             
24 |         serde::de::Deserialize::deserialize(self.metadata)                                                                  
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `bom::_::_serde::Deserializer<'_>` is not implemented for `toml::map::Map<std::string::String, Value>`
```

That's why I need to use `.try_into` and convert it into a `toml::Value`.

Reply:
Ah, that is because you're trying to work with the `serde` primitives that don't know anything about `TOML`. Since `Deserializer` is not implemented for `Table`, serde can't read its primitives from our `Table`.

All toml `Value`s have `try_into()` which converts them to a `T` if `Deserialize` is implemented for `T`. Sadly, the `Table` type we use in `Entry` is not a `Value` so we need to make it one by wrapping it. As outlined earlier, we can't get around the clone since we're only borrowing the `metadata`.

Long story short, I think the code you have in here is perfectly fine.